ee/katc/sqlite.go

```diff
@@ -0,0 +1,84 @@
+package katc
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log/slog"
+	"path/filepath"
+
+	_ "modernc.org/sqlite"
+)
+
+// sqliteData is the dataFunc for sqlite KATC tables
+func sqliteData(ctx context.Context, pathPattern string, query string, columns []string, slogger *slog.Logger) ([]map[string][]byte, error) {
+	sqliteDbs, err := filepath.Glob(pathPattern)
+	if err != nil {
+		return nil, fmt.Errorf("globbing for files with pattern %s: %w", pathPattern, err)
+	}
+
+	results := make([]map[string][]byte, 0)
+	for _, sqliteDb := range sqliteDbs {
+		resultsFromDb, err := querySqliteDb(ctx, sqliteDb, query, columns, slogger)
+		if err != nil {
+			return nil, fmt.Errorf("querying %s: %w", sqliteDb, err)
+		}
+		results = append(results, resultsFromDb...)
+	}
+
+	return results, nil
+}
+
+// querySqliteDb queries the database at the given path, returning rows of results
+func querySqliteDb(ctx context.Context, path string, query string, columns []string, slogger *slog.Logger) ([]map[string][]byte, error) {
+	conn, err := sql.Open("sqlite", path)
```

Comment:
We should open RO

Reply:
I don't think there's an open read-only option, unless I'm missing it

Reply:
Thank you for finding that -- I'll give it a shot!

Reply:
It's also an example of the URL style :P